### PR TITLE
Only push tagged to test.pypi.org

### PR DESCRIPTION
**Pull request by dmurphy18**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: Release
 
 on:
   release:
-    types: [created]
+    types: [published]
```
Comment (ScriptAutomate) [inline .github/workflows/release.yml:5]:
```suggestion
    types: [released]
```

Ref: https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#release

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -29,6 +29,7 @@ jobs:
         nox -e build
 
     - name: Publish package distributions to PyPI
+      if: github.event.release.released
```
Comment (ScriptAutomate) [inline .github/workflows/release.yml:32]:
```suggestion
```

If `types: [released]` works (from other suggestion), we can remove this if conditional.

File: .github/workflows/test_publish.yml [hunk 3]
```diff
@@ -2,9 +2,8 @@
 name: Test PyPI Publish
 
 on:
-  push:
-    branches:
-      - main
+  release:
+    types: [published]
```
Comment (ScriptAutomate) [inline .github/workflows/test_publish.yml:6]:
```suggestion
    types: [prereleased]
```

Ref: https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#release

File: .github/workflows/test_publish.yml [hunk 4]
```diff
@@ -31,6 +30,7 @@ jobs:
         nox -e build
 
     - name: Publish distribution 📦 to Test PyPI
+      if: github.event.release.prereleased
```
Comment (ScriptAutomate) [inline .github/workflows/test_publish.yml:33]:
```suggestion
```

If `types: [prereleased]` works (from other suggestion), we can remove this if conditional.

**Review by ScriptAutomate — CHANGES_REQUESTED** [review]

**Comment by ScriptAutomate** [comment]
If my suggestions are adopted, it will mean:
- To publish to test pypi: GitHub Releases -> Draft a new release -> create a new tag (with `rc` naming convention) -> check **"Set as a pre-release"** -> publish release
- To publish to prod pypi: GitHub Releases -> Draft a new release -> create a new tag (with `rc` naming convention) -> check **"Set as the latest release"** -> publish release

> **NOTE:** GitHub Releases do NOT do annotated tags. On GitHub, you can see what user did a Release, if needed. You do lose out on some `git` functionality without being able to have the annotated tags, though, with timestamps, `git describe`, user, signing, etc.
>
> Feature request to GitHub: https://github.com/orgs/community/discussions/4924
>
> A workaround would be to figure out how to have the GitHub workflows triggered by prereleases and releases to delete and then recreate the tag but as an annotated tag, if an annotated tag would be desired. That, or have a workflow you can launch that will create an annotated tag, first, before going through the Release page and doing a release based on the new tag. A variant of this could prob be done following https://stackoverflow.com/a/64479344/5340149 (but with a prior step doing `github.rest.git.createTag()` which would be required for an annotated tag before the step doing `github.rest.git.createRef()`.

File: .github/workflows/release.yml (hunk 1 above)
Comment (dmurphy18) [reply .github/workflows/release.yml]: done

File: .github/workflows/test_publish.yml (hunk 3 above)
Comment (dmurphy18) [reply .github/workflows/test_publish.yml:6]: done

File: .github/workflows/release.yml (hunk 2 above)
Comment (dmurphy18) [reply .github/workflows/release.yml]: Done

File: .github/workflows/test_publish.yml (hunk 4 above)
Comment (dmurphy18) [reply .github/workflows/test_publish.yml]: done

**Review by ScriptAutomate — APPROVED** [review]

**Pull request merged**